src/parquet/arrow/arrow-schema-test.cc

```diff
@@ -807,5 +807,17 @@ TEST_F(TestConvertArrowSchema, ParquetFlatDecimals) {
   CheckFlatSchema(parquet_fields);
 }
 
+TEST(InvalidSchema, ParquetNegativeDecimalScale) {
+  const auto& type = ::arrow::decimal(23, -2);
+  const auto& field = ::arrow::field("f0", type);
+  const auto& arrow_schema = ::arrow::schema({field});
```

Comment:
I don't believe the `&` has an effect here

Reply:
Actually this could also be risky. The return values here actually return an object someone needs to own, by only storing a reference, the ownership of the objects is completely open.